module/connection/ServerDialog.kt

```diff
@@ -78,48 +91,80 @@ object ServerDialog {
-                    caCertificate.isBlank() -> StudioState.client.tryConnectToTypeDBClusterAsync(
-                        address, username, password, tlsEnabled
-                    ) { StudioState.client.connectServerDialog.close() }
-                    else -> StudioState.client.tryConnectToTypeDBClusterAsync(
-                        address, username, password, caCertificate
-                    ) { StudioState.client.connectServerDialog.close() }
+                TYPEDB ->  {
+                    StudioState.client.tryConnectToTypeDBAsync(coreAddress) {
+                        StudioState.client.connectServerDialog.close()
+                    }
+                }
+                TYPEDB_CLUSTER -> {
+                    when {
+                        caCertificate.isBlank() -> StudioState.client.tryConnectToTypeDBClusterAsync(
+                            clusterAddresses.toSet(), username, password, tlsEnabled
+                        ) {
+                            StudioState.client.connectServerDialog.close()
+                        }
+                        else -> StudioState.client.tryConnectToTypeDBClusterAsync(
+                            clusterAddresses.toSet(), username, password, caCertificate
+                        ) {
+                            StudioState.client.connectServerDialog.close()
+                        }
+                    }
                 }
             }
             appData.server = server
-            appData.address = address
+            appData.coreAddress = coreAddress
+            appData.clusterAddresses = clusterAddresses
             appData.username = username
             appData.tlsEnabled = tlsEnabled
             appData.caCertificate = caCertificate
         }
     }
 
+    private object CreateAddressForm : Form.State {
+
+        var value: String by mutableStateOf("")
+
+        override fun cancel() {
+            StudioState.client.manageAddressesDialog.close()
+        }
+
+        override fun isValid(): Boolean {
+            return value.isNotBlank()
+        }
+
+        override fun trySubmit() {
+            assert(isValid())
+            state.clusterAddresses.add(value)
+            value = ""
+        }
+    }
+
     @Composable
     fun MayShowDialogs() {
         if (StudioState.client.connectServerDialog.isOpen) ConnectServer()
+        if (StudioState.client.manageAddressesDialog.isOpen) ManageAddresses()
     }
 
     @Composable
     private fun ConnectServer() {
-        val state = remember { ConnectServerForm() }
         Dialog.Layout(StudioState.client.connectServerDialog, Label.CONNECT_TO_TYPEDB, WIDTH, HEIGHT) {
             Submission(state = state, modifier = Modifier.fillMaxSize(), showButtons = false) {
                 ServerFormField(state)
-                AddressFormField(state, StudioState.client.isDisconnected)
                 if (state.server == TYPEDB_CLUSTER) {
+                    ClusterAddressesFormField(state)
```

Comment:
We decided not to focus the field itself, because that's pointless, you can't type into it. But might it make sense to focus the + button so you can hit ENTER to immediately begin adding addresses?

Reply:
Done. Lots of fast keyboard navigation possible now.